atomata/capstone_WebClient
# Test overlayfunc

### Summary 
Overall, there are new tests for each function. Also, for each function, hook, and const, the types are annotated. 

### Details

- Overlay: Some functions and the hooks are moved into the util/overlayfunc/overlayfunc.tsx. The test file is overlay.test

- PreviewOverlay: removed hook functions into util/previewOverlayFun/previewOverlayfunc.tsx

- Types for ActionBox, ActionSequenceBox and ApparatusListBox aren't finalized yet. However, they do not affect how the webpage functions. 



### How to test?
- To run the test and the test coverage, please check the readme file.
- Check if all functionalities still run the way they are supposed to.

### Checks
